src/dialogs/examples/teams/UpdateCardMsgDialog.ts

```diff
@@ -6,49 +6,49 @@ import { Strings } from "../../../locale/locale";
 let config = require("config");
 
 export class UpdateCardMsgDialog extends TriggerActionDialog {
-
+    // update the card, if user has already setup the card message from below dialog file using update button click
+    // microsoft-teams-sample-complete-node\src\dialogs\examples\teams\UpdateCardMsgSetupDialog.ts
     private static async updateCardMessage(session: builder.Session, args?: any | builder.IDialogResult<any>, next?: (args?: builder.IDialogResult<any>) => void): Promise<void> {
-        if (session.conversationData.updateCardCounter !== null) {
-            session.conversationData.updateCardCounter = session.conversationData.updateCardCounter + 1;
-        }
-
-        if (args.address) {
+        if (session.message.replyToId)
+        {
             let buttons = new Array<builder.CardAction>();
+            let updateCardCounter = session.message.value.updateCounterKey;
+            let messageBackButtonValue = JSON.stringify({ updateCounterKey: ++updateCardCounter });
 
-            buttons.push(new builder.CardAction(session)
-                .type("invoke")
-                .title(Strings.update_card_button, session.conversationData.updateCardCounter)
-                .value("{" +
-                    "\"dialog\": \"" + DialogIds.UpdateCardMsgDialogId + "\", " +
-                    "\"address\": " + JSON.stringify(args.address) + "" +
-                "}"),
-            );
+            let messageBackButton = builder.CardAction.messageBack(session, messageBackButtonValue)
+                .displayText(Strings.messageBack_button_display_text)
+                .title(Strings.update_card_button, updateCardCounter)
```

Comment:
Can we keep this as always "Update card", and instead change the card title to be "Update #{updateCount}"?

Reply:
okay, then I have to do the same changed in C# sample as well to keep the consistency.